### PR TITLE
Wip/drothe image support

### PR DESCRIPTION
**Pull request by d-rothe**
4 pull requests belonging together (same branch name): DeviceAccess, ControlSystemAdapter, DoocsAdapter, DoocsBackend.

Note: 
I decided against 'deprecated' attribute in MappedStruct in favor of documentation note that we discourage MappedStruct use.  'deprecated' would not be precise since we do not have a better solution yet. Also, compilation warnings in DeviceAccess will be only nuisance and not helpful.

### CODE REVIEW

**Comment by mhier** [comment]
Formal comment about commit messages:
Commit messages are not following conventional commits. Also the commit [replace 'deprecated' tag by documentation note that we discourage Map…](https://github.com/ChimeraTK/DeviceAccess/pull/368/commits/3c95fa7239c817bdc140c76ff15f106646625112) does not replace anything (just adds a comment)...

**Comment by mhier** [comment]
I assume that the first two commits are just importing the mentioned classes without modification, hence I reviewed only the other commits which seem fine. Please modify the commit messages and then merge without squashing.

**Review by mhier — APPROVED** [review]

**Review by killenb — APPROVED** [review]
Please check if toString and toDouble can be replaced with something existing from the util section. If yes, use them and remove the two files. If  not, move these two rather specific conversion helpers to the test code.

File: util/include/toDouble.h
[the remark this answers is not in the document]
Comment (d-rothe) [reply util/include/toDouble.h:1]: I agree. I got rid of 'toDouble.h' and 'toType.h' here and moved them back into ControlSystemAdapter tests from where they originate.

File: util/include/TypeChangingDecorator.h [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+   * has it's own buffer of type UserType and synchronises it in the preWrite()
+   * and postRead() functions with the implementation. You don't have to care
+   * about the implementation type of the transfer element. The factory will
+   * automatically create the correct decorator.
+   * 
+   * Note: it is possible to obtain multiple decorators of different types for the same accessor. The user needs to
+   * ensure that the preXxx/postXxx transfer functions are properly called for all decorators when required.
+   *
+   *  @param transferElement The TransferElement to be decorated. It can either be
+   * an NDRegisterAccessor (usually the case) or and NDRegisterAccessorBridge (but
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply util/include/TypeChangingDecorator.h]: fixed comment

File: util/include/toType.h
[the remark this answers is not in the document]
Comment (d-rothe) [reply util/include/toType.h:1]: same reply

**Pull request merged**